greedy_methods/kruskal.py

```diff
@@ -0,0 +1,120 @@
+"""
+https://en.wikipedia.org/wiki/Kruskal%27s_algorithm
+"""
+
+import doctest
+from collections.abc import Iterable
+from dataclasses import dataclass
+
+
+@dataclass
+class Edge:
+    """
+    Represents an edge in a graph.
+
+    Attributes:
+        start (str): The starting vertex of the edge.
+        end (str): The ending vertex of the edge.
+        weight (int): The weight of the edge.
```

Comment:
Please do not repeat the datatypes which are verified by mypy below because readers will become confused if one is changed but the other is not.
```suggestion
        start: The starting vertex of the edge.
        end: The ending vertex of the edge.
        weight: The weight of the edge.
```